backend.native/tests/objcexport/values.kt

```diff
@@ -1012,3 +1011,30 @@ fun mutULong2Long(): MutableMap<ULong, Long> = mutableMapOf(Pair(0x8000_0000_000
 fun mutFloat2Float(): MutableMap<Float, Float> = mutableMapOf(Pair(3.14f, 100f))
 fun mutDouble2String(): MutableMap<Double, String> = mutableMapOf(Pair(2.718281828459045, "2.718281828459045"))
 
+@OptIn(ExperimentalAssociatedObjects::class)
+@AssociatedObjectKey
+@Retention(AnnotationRetention.BINARY)
+annotation class Associated_Kt35940(val kClass: KClass<*>)
+
+private class CKt35940(var list: List<String>?)
+
+private interface IKt35940 {
+    fun bar(c: CKt35940)
+}
+
+private object IKt35940Impl : IKt35940 {
+    override fun bar(c: CKt35940) {
+        c.list = mutableListOf("zzz")
+    }
+}
+
+@Associated_Kt35940(IKt35940Impl::class)
+private class IKt35940ImplHolder
+
+@OptIn(ExperimentalAssociatedObjects::class)
+fun testKt35940(): String {
```

Comment:
What exactly does it test?
I tried removing the code for associated objects from `Devirtualization`, but the test still passes with `-opt`.

Reply:
Hmm, it's been copied from the test `associatedObjects1`, may be I've copied wrong part.

Reply:
Weird, if I move the test to a separate file and compile it to a `kexe`, then it fails as expected.

Reply:
This probably means that the test is not self-contained enough. IIRC, it checks that `ArrayList` is in instantiating classes.

Reply:
Actually, the call site hasn't been devirtualized, I'll dig deeper to find out why.